tutorials-and-examples/nvidia-nim/blueprints/drugdiscovery/test-generative-virtual-screening.py

```
@@ -0,0 +1,110 @@
+import requests
+
+AF2_HOST = "http://localhost:8010"
+MOLMIM_HOST = "http://localhost:8011"
+DIFFDOCK_HOST = "http://localhost:8012"
+
+
+def check_readiness(service_name, url):
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an exception for bad status codes
+        print(f"{service_name} readiness check successful at {url}.")
+        return True
+    except requests.exceptions.RequestException as e:
+        print(f"{service_name} readiness check failed: {e}")
+        return False
+
+
+check_readiness("AlphaFold2", f"{AF2_HOST}/v1/health/ready")
```

These calls don't check the return status from `check_readiness`. The first time I ran this I only had one port-forward command running and two of the checks failed but the program kept executing. 

Alternatively, the check_readiness function could exit if it gets an exception instead of swallowing it and returning a boolean.

---

removed due to repeat in yaml